test/mock_http_client.go

```diff
@@ -85,11 +85,16 @@ func NewMockHttpClient(mocks []MockResult) rest.HTTPClient {
 
 // getPath returns the path from the provided URL.
 func getPath(url *url.URL) string {
+	var path string
 	if url.RawPath != "" {
-		return url.RawPath
+		path = url.RawPath
 	} else {
-		return url.Path
+		path = url.Path
 	}
+	if url.RawQuery != "" {
```

Comment:
updated.

Reply:
We are using `RawQuery` instead of `RawPath` in the second if. The raw query returns anything thats after the `?` symbol as an URL encoded string. This allows us to do mock matching by query params, which then allows us to write tests for the full URL, instead of only path.

Reply:
Oh sorry my bad, probably lazy eyes after vacation.

Ok I get it now, but, why do we need it for test? is this a test thing only? Is it used in real case examples? I am confused.

Reply:
Yes, in the sonarqube plugin in basically all of the APIs calls we have conditional query params, which is not testable with current mocking behavior.

Reply:
Thanks